helm/cluster-openstack/values.yaml

```diff
@@ -1,41 +1,32 @@
-cloudName: openstack  # Name of cloud in cloud-config secret "cloud.yaml" value.
-baseDomain: "example.com"
-managementCluster: ""
-
-kubernetesVersion: 1.20.9
-releaseVersion: 20.0.0-alpha1
-
-nodeCIDR: ""  # The OpenStack Subnet to be created.
-externalNetworkID: ""  # UUID of external network (optional).
+clusterName: ""  # Name of cluster. Used as base name for all cluster resources in the management cluster.
+dnsNameservers: []  # IPs of DNS nameservers to be used by the cluster machines (optional, will use OS defaults if not specified).
+externalNetworkID: ""  # UUID of external network (optional unless there are multiple external networks).
+kubernetesVersion: ""  # Version of Kubernetes. If machine images are specified, this will only be used for metadata.
+managementCluster: ""  # Name of cluster which owns this workload cluster.
+nodeCIDR: ""  # The CIDR of the OpenStack subnet to be created for cluster machines.
+organization: ""  # Organization which owns the cluster.
 
 oidc:
-  enabled: false
+  enabled: false  # If true, flags will be configured on the control plane for OIDC-based SSO authentication.
 
 bastion:
-  flavor: n1.tiny
-  image: ""  # Ignored if rootVolume.sourceUUID is set (optional).
-  rootVolume:
-    diskSize: 10
-    sourceUUID: ""  # UUID of the bastion image (optional).
-
-rootVolume:
-  enabled: false
-  sourceUUID: ""  # UUID of root volume source image (optional).
-
-nodeClasses:
-- name: default  # Name of class.
-  machineFlavor: n1.small  # Flavor of machine.
-  diskSize: 50  # Root volume disk size in GB.
+  diskSize: 0  # Root volume disk size in GB. Ignored if bootFromVolume is false. Should be at least as large as the source image.
+  flavor: ""  # Project-dependent flavor of bastion machine.
+  image: ""  # Image name or root volume source UUID if bootFromVolume is true.
+  bootFromVolume: false  # If true, machines will use a persistent root volume instead of an ephemeral volume.
 
 controlPlane:
-  class: default  # Name of node class to be used by control plane nodes.
-  replicas: 1  # Number of replicas in control plane. Should be an odd number.
-  availabilityZones: []
+  availabilityZones: []  # Unique availability zones for control plane nodes. If specified, should have one item or one for each replica (optional).
+  diskSize: 0  # Root volume disk size in GB. Ignored if bootFromVolume is false. Should be at least as large as the source image.
+  flavor: ""  # Project-dependent flavor of control plane machines.
+  image: ""  # Image name or root volume source UUID if bootFromVolume is true.
+  replicas: 0  # Number of replicas in control plane. Should be an odd number.
+  bootFromVolume: false  # If true, machines will use a persistent root volume instead of an ephemeral volume.
+
+nodeClasses: []  # Class definitions for worker node pools. In addition to a required "name", may include "diskSize", "flavor", "image", and "bootFromVolume" as defined for control plane and bastion above.
```

Comment:
I'd prefer the old approach of specifying fully one exemple entry. Or maybe two.

Reply:
I agree that it's nice to know what the structure looks like, but I don't like putting example values because it's installation specific. The user must provide values that work for their environment.